### PR TITLE
Support provisioning v2 backup

### PR DESCRIPTION
**Pull request by StrongMonkey**

### CODE REVIEW

File: pkg/controllers/restore/controller.go [hunk 1]
```diff
@@ -483,16 +483,26 @@ func customize(obj *unstructured.Unstructured) {
 		logrus.Debugf("Secrets section is ignored in ServiceAccount %s/%s", obj.GetNamespace(), obj.GetName())
 	case "Cluster":
 		// for fleet cluster it needs to be reimported in order to reissue service account token that is no longer valid
-		if obj.GetAPIVersion() == "fleet.cattle.io/v1alpha1" {
+		switch obj.GetAPIVersion() {
+		case "fleet.cattle.io/v1alpha1":
 			// only warn error if field can't be found. In case there are breaking api changes error will be printed as Warn and user has workaround to patch it.
-			redeployAgentGeneration, _, err := unstructured.NestedInt64(obj.Object, "spec", "redeployAgentGeneration")
+			redeployAgentGeneration, _, err := unstructured.NestedFloat64(obj.Object, "spec", "redeployAgentGeneration")
```
Comment (StrongMonkey) [inline pkg/controllers/restore/controller.go:489]: Note: I found out during testing that this has to be float64 due the json decoding.

**Comment by StrongMonkey** [comment]
https://github.com/rancher/rancher/issues/34107

**Comment by StrongMonkey** [comment]
@snasovich This is ready to be reviewed with https://github.com/rancher/rancher/pull/34181.

**Review by jiaqiluo — APPROVED** [review]
LGTM

File: charts/rancher-backup/files/default-resourceset-contents/provisioningv2.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+- apiVersion: "provisioning.cattle.io/v1"
+  kindsRegexp: "."
+- apiVersion: "rke-machine-config.cattle.io/v1"
+  kindsRegexp: "."
+- apiVersion: "rke-machine.cattle.io/v1"
+  kindsRegexp: "."
+- apiVersion: "rke.cattle.io/v1"
+  kindsRegexp: "."
+- apiVersion: "apiextensions.k8s.io/v1"
```
Comment (snasovich) [inline charts/rancher-backup/files/default-resourceset-contents/provisioningv2.yaml:9]: Suggesting to have rule for CRD to be the first one. It doesn't change anything functionally, but this way the configuration is a bit more logical and easier to read.

**Review by snasovich — APPROVED** [review]
Would be great if you could address one comment I've added - that is if you agree with it.

**Review by snasovich — APPROVED** [review]
LGTM, we should be all set as you have Jack's approval.

**Comment by StrongMonkey** [comment]
Context of this PR: https://github.com/rancher/rancher/issues/34107#issuecomment-899820635

**Pull request merged**